tests/forecast/test_boostingtrees.py

```diff
@@ -65,46 +64,97 @@ def __init__(self, *args, **kwargs):
             )
         )
 
-    def test_forecast_multi(self):
+        self.model_sequence = LGBMForecaster(
+            LGBMForecasterConfig(
+                max_forecast_steps=self.max_forecast_steps,
+                maxlags=self.maxlags,
+                target_seq_index=self.i,
+                prediction_stride=self.max_forecast_steps,
+                n_estimators=20,
+                max_depth=5,
+                n_jobs=1,
+                transform=MinMaxNormalize(),
+                invert_transform=True,
+            )
+        )
+
+    def test_forecast_multi_autoregression(self):
         logger.info("Training multivariate model...")
-        yhat, _ = self.model.train(self.train_data)
+        yhat, _ = self.model_autoregression.train(self.train_data)
 
         # Check RMSE with multivariate forecast inversion
-        forecast, _ = self.model.forecast(self.max_forecast_steps)
+        forecast, _ = self.model_autoregression.forecast(self.max_forecast_steps)
         rmse = ForecastMetric.RMSE.value(self.test_data, forecast, target_seq_index=self.i)
         logger.info(f"Immediate forecast RMSE: {rmse:.2f}")
         # self.assertAlmostEqual(rmse, 2.9, delta=0.1)
 
         # Check look-ahead RMSE using time_series_prev
-        testing_data_gen = gen_next_seq_label_pairs(self.test_data, self.i, self.maxlags, self.max_forecast_steps)
-        testing_instance, testing_label = next(testing_data_gen)
-        pred, _ = self.model.forecast(testing_label.time_stamps, testing_instance)
+        dataset = RollingWindowDataset(self.test_data,
+                                       self.i,
+                                       self.maxlags,
+                                       self.max_forecast_steps,
+                                       ts_index=True)
+        testing_instance, testing_label = next(iter(dataset))
+        pred, _ = self.model_autoregression.forecast(testing_label.time_stamps, testing_instance)
         lookahead_rmse = ForecastMetric.RMSE.value(testing_label, pred, target_seq_index=self.i)
         logger.info(f"Look-ahead RMSE with time_series_prev: {lookahead_rmse:.2f}")
         # self.assertAlmostEqual(lookahead_rmse, 18.9, delta=0.1)
 
         # save and load
-        self.model.save(dirname=join(rootdir, "tmp", "lgbmforecaster"))
+        self.model_autoregression.save(dirname=join(rootdir, "tmp", "lgbmforecaster"))
+        loaded_model = LGBMForecaster.load(dirname=join(rootdir, "tmp", "lgbmforecaster"))
+        loaded_pred, _ = loaded_model.forecast(testing_label.time_stamps, testing_instance)
+        self.assertEqual(len(loaded_pred), self.max_forecast_steps)
+        self.assertAlmostEqual((pred.to_pd() - loaded_pred.to_pd()).abs().max().item(), 0, places=5)
+
+    def test_forecast_multi_sequence(self):
```

Comment:
Maybe rename this to `test_forecast_multi_seq2seq`?

Reply:
absolutely

Reply:
Done